### ISSUE TITLE
Uninstalling v1.3 does not delete UPnP rules

### ISSUE THREAD

**Issue by yahaverlich** [moonlight-stream/Internet-Streaming-Helper#1]
I installed v1.3 of Moonlight Internet Streaming Helper ( I have never installed a previous version). After I could not get streaming to work, I manually forwarded ports, and uninstalled this Helper, but the UPnP rules still exist and conflict with my manual port forwarding. Therefore, I must disable UPnP for my streaming over the internet to work.

This references the original issue ticket that I put in the wrong place:
https://github.com/moonlight-stream/moonlight-qt/issues/117

Here ismy C:/ProgramData/MISS log, with the WAN address blanked out:

[miss-current.log](https://github.com/moonlight-stream/Internet-Streaming-Helper/files/2522462/miss-current.log)

**Comment by cgutman** [comment]
It looks like it deleted the rules before stopping.

Can you screenshot the UPnP page in your router showing the active rules?

**Comment by yahaverlich** [comment]
<img width="1043" alt="screen shot 2018-10-27 at 11 43 56 pm" src="https://user-images.githubusercontent.com/10982729/47612940-35328480-da42-11e8-8641-dd8bca06b55c.png">

The external and internal ports don't match, which I think are the wrong rules. Regardless, these are the same ports that Moonlight needs to use so it conflicts with my manual forwarding. I also don't think any other program on my pc is creating these rules, it has to be coming from Moonlight

**Comment by yahaverlich** [comment]
This error is the reason I could not get the Internet Streaming Helper to work. The Helper created all 8 of the correct rules with internal and external ports matching, alongside these 6 rules that conflicted. That is why manual forwarding with UPnP off worked for me

**Comment by cgutman** [comment]
OK, as I suspected, those rules are definitely GeForce Experience's, not Moonlight's. We never create a rule where the internal and external ports differ. You can see the rules GFE is creating in C:\ProgramData\NVIDIA Corporation\nvstreamsvc\nssCurrent.log by searching for "GSPortForward" or "UPnP"

There's a description field also sent when UPnP rules are created that your router doesn't seem to show, but you can see from mine that those are GFE's rules. Note how GFE only creates UDP rules and how the external port never matches the internal port, but the Moonlight rules always match:
<img width="905" alt="screen_shot_2018-10-26_at_7 31 34_pm" src="https://user-images.githubusercontent.com/2695644/47613058-f651fe00-da44-11e8-95d3-03c1a06afa80.png">

**Comment by yahaverlich** [comment]
Thanks for the rundown! Seems it was not a bug and I should have probably discussed this in the Discord instead. 

Is the UPnP IGD violation the reason I cannot stream over internet when UPnP is on, or is it simply a router being buggy? Because no matter what I try, control stream establishment will fail if I use Internet Streaming Helper or if UPnP is turned on while I do manual port forwarding. However, the combination of manual port forwarding and UPnP turned off, will let me stream over the internet successfully. Is there any way I can fix this?

**Comment by yahaverlich** [comment]
Ok, so today I went to go follow the instructions you gave above. Lo and behold, I was able to stream over 4G network without making any changes. I wanted to test it out one more time before I did anything and it worked. I was able to stream using manual port forwarding, while still having my UPnP turned on (and all of GFE's rules still there). Therefore, I don't think I need to prevent GFE from creating those UPnP rules (what would happen if I made GFE stop making those rules like you mentioned above, other than preventing regular GameStream from working?). 

Soon I will go ahead and try using the Internet Streaming Helper without manual port forwarding to verify if it works, but I think it will. It seems it must have been something buggy with my network or my router; it is weird that turning UPnP off solved the issue previously, and now it works with it on.

My router is a Netgear Orbi RBR50. Thanks for the help! I'll update soon to see if MISH works as is, and if it doesn't I will try deleting GFE's UPnP rules and I will try again.
